### PR TITLE
check for a given ns istio install before warn

### LINKED ISSUE #28775: External istiod install generates an confusing warning
**Bug description**

Following up [external istiod doc](https://preliminary.istio.io/latest/docs/setup/additional-setup/external-controlplane/):

```
$ istioctl install -f external-istiod.yaml --set tag=latest --set hub=gcr.io/istio-testing --revision=external
! Istio is being upgraded from 1.8.0 -> 1.9.0.
! Before upgrading, you may wish to 'istioctl analyze' to check for IST0002 deprecation warnings.
This will install the Istio 1.9.0 profile into the cluster. Proceed? (y/N) n
Cancelled.
```

or

```
$ istioctl install -f external-istiod.yaml --set tag=latest --set hub=gcr.io/istio-testing                              
! Istio control planes installed: 1.8.0.
! Use --revision or --force to install Istio.
This will install the Istio 1.9.0 profile into the cluster. Proceed? (y/N) ^C
```

The warning message here is super confusing and should not be necessary because external-istiod.yaml is installing istiod in a different ns from default istio-system so no conflicts.

[ ] Docs
[ x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
Would not expect such a warning when installing to a different ns.

cc @rvennam

### PR DESCRIPTION
**Pull request by linsun**
fixes: #28775

[ ] Configuration Infrastructure
[ ] Docs
[ x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ x] Does not have any changes that may affect Istio users. - Think this behavior is not released in 1.8.

### CODE REVIEW

File: operator/cmd/mesh/install.go [hunk 1]
```diff
@@ -318,5 +320,5 @@ func getProfileAndEnabledComponents(setOverlay []string, inFilenames []string, f
 			}
 		}
 	}
-	return profile, enabledComponents, nil
+	return profile, iop.Namespace, enabledComponents, nil
```
Comment (morvencao) [inline operator/cmd/mesh/install.go:323]: I thought the returned namespace should be the namespace in which istio control plane will be installed, right?

Comment (linsun) [reply operator/cmd/mesh/install.go]: @morvencao thanks!  I updated the code to use istioNamespace, let me know if it makes sense.

**Comment by linsun** [comment]
Did a bit testing locally, seems to be working okay without generating the warning in this case.

```
$ go run "${GOPATH}"/src/istio.io/istio/istioctl/cmd/istioctl/main.go install -f  ../testingdir/external-istiod.yaml --set tag=latest --set hub=gcr.io/istio-testing --charts manifests 
This will install the Istio 1.9.0 default profile with ["Istiod"] components into the cluster. Proceed? (y/N) 
```

File: operator/cmd/mesh/install.go [hunk 2]
```diff
@@ -318,5 +321,16 @@ func getProfileAndEnabledComponents(setOverlay []string, inFilenames []string, f
...
+	globalI := iop.Spec.Values["global"]
+	global, ok := globalI.(map[string]interface{})
+	if !ok {
+		return profile, controller.IstioNamespace, enabledComponents, nil
+	}
+	ns, ok := global["istioNamespace"].(string)
+	if !ok {
+		return profile, controller.IstioNamespace, enabledComponents, nil
+	}
+	return profile, ns, enabledComponents, nil
```
Comment (morvencao) [inline operator/cmd/mesh/install.go:335]: @linsun I think you can reuse the function: https://github.com/istio/istio/blob/cbe5f1108f10ba3803e5ac3a6ae1a8e35e26531c/operator/pkg/apis/istio/v1alpha1/common.go#L30

Comment (linsun) [reply operator/cmd/mesh/install.go]: Ah thanks @morvencao!

**Comment by linsun** [comment]
Close this PR to open a new one due to bad merge.

**Pull request closed without merging**